driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -411,14 +430,12 @@ private Connection findLeastBusyForShard(int shardId) {
       if (host.convictionPolicy.canReconnectNow()) {
         if (connectionsPerShard == 0) {
           maybeSpawnNewConnection(shardId);
-          return enqueue(timeout, unit, maxQueueSize, shardId);
```

Comment:
Why was it removed?

Reply:
The pick a connection to another shard instead of waiting for the "right" connection while it is being established.